### PR TITLE
Vite migration

### PR DESCRIPTION
**Pull request by HieronymusLex**
*Issue #, if available:*
N/A
*Description of changes:*
Migrate to Vite

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: cli.js [hunk 1]
```diff
@@ -60,7 +60,7 @@ const generateConfig = (options) => {
 
   getStackDetails(samConfig, samConfigEnv)
     .then(details => getOutputs(details.stackName, details.region))
-    .then(stackOutputs => writeConfigFile(".env.template", ".env.local", stackOutputs))
+    .then(stackOutputs => writeConfigFile(".env.template", ".env", stackOutputs))
```
Comment (sprkem) [inline cli.js:63]: Why this change? I used .env.local successfully with Vite elsewhere.

File: vite.config.js [hunk 2]
```diff
@@ -0,0 +1,17 @@
+import { defineConfig } from "vite";
+import react from "@vitejs/plugin-react";
+
+// https://vitejs.dev/config/
+export default defineConfig({
+  plugins: [react()],
+  define: {
+    global: {},
+    "process.env": process.env,
```
Comment (sprkem) [inline vite.config.js:9]: You might be able to get away without this. I was trying to retain the format of constants.js but it wouldn't work.

File: cli.js (hunk 1 above)
Comment (HieronymusLex) [reply cli.js]: Good point - updated

File: vite.config.js (hunk 2 above)
Comment (HieronymusLex) [reply vite.config.js]: yeah seems to work without it

**Review by sprkem — APPROVED** [review]
